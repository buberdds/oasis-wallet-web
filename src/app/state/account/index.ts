import { PayloadAction } from '@reduxjs/toolkit'
import { Transaction } from 'app/state/transaction/types'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { AccountState, Account } from './types'

export const initialState: AccountState = {
  address: '',
  liquid_balance: 0,

  accountError: null,
  transactions: [],
  transactionsError: null,
  loading: true,
}

const slice = createSlice({
  name: 'account',
  initialState,
  reducers: {
    clearAccount(state, action: PayloadAction<void>) {
      Object.assign(state, initialState)
    },
    fetchAccount(state, action: PayloadAction<string>) {},
    accountLoaded(state, action: PayloadAction<Account>) {
      state.accountError = null
      Object.assign(state, action.payload)
    },
    accountError(state, action: PayloadAction<string>) {
      state.accountError = action.payload
    },
    transactionsLoaded(state, action: PayloadAction<Transaction[]>) {
      state.transactionsError = null
      state.transactions = action.payload
    },
    transactionsError(state, action: PayloadAction<string>) {
      state.transactionsError = action.payload
    },
    setLoading(state, action: PayloadAction<boolean>) {
      state.loading = action.payload
    },
  },
})

export const { actions: accountActions } = slice

export default slice.reducer
