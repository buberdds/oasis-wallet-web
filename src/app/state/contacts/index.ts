import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { ContactsState, Contact } from './types'

export const initialState: ContactsState = {}

const slice = createSlice({
  name: 'contacts',
  initialState,
  reducers: {
    add(state, action: PayloadAction<Contact>) {
      state[action.payload.address] = action.payload
    },
    update(state, action: PayloadAction<Contact>) {
      state[action.payload.address] = action.payload
    },
    delete(state, action: PayloadAction<string>) {
      delete state[action.payload]
    },
  },
})

export const { actions: contactsActions } = slice

export default slice.reducer
