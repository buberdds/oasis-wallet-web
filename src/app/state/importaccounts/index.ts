import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { ImportAccountsListAccount, ImportAccountsState, ImportAccountsStep } from './types'

export const initialState: ImportAccountsState = {
  accounts: [],
  deviceRequested: false,
  showAccountsSelectionModal: false,
}

const slice = createSlice({
  name: 'importAccounts',
  initialState,
  reducers: {
    clear(state, action: PayloadAction<void>) {
      state.accounts = []
      state.error = undefined
      state.step = undefined
      state.showAccountsSelectionModal = false
    },
    deviceRequested(state, action: PayloadAction<void>) {
      state.deviceRequested = true
    },
    enumerateAccountsFromLedger(state, action: PayloadAction<void>) {
      state.step = undefined
      state.accounts = []
      state.showAccountsSelectionModal = true
    },
    enumerateAccountsFromMnemonic(state, action: PayloadAction<string>) {
      state.step = undefined
      state.accounts = []
      state.showAccountsSelectionModal = true
    },
    toggleAccount(state, action: PayloadAction<number>) {
      const index = action.payload
      state.accounts[index].selected = !state.accounts[index].selected
    },
    accountsListed(state, action: PayloadAction<ImportAccountsListAccount[]>) {
      state.accounts = action.payload
    },
    setStep(state, action: PayloadAction<ImportAccountsStep>) {
      state.step = action.payload
    },
    operationFailed(state, action: PayloadAction<ErrorPayload>) {
      state.error = action.payload
      state.step = undefined
    },
  },
})

export const { actions: importAccountsActions } = slice

export default slice.reducer
