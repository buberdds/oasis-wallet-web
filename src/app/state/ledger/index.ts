import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { LedgerAccount, LedgerState, LedgerStep } from './types'

export const initialState: LedgerState = { accounts: [] }

const slice = createSlice({
  name: 'ledger',
  initialState,
  reducers: {
    clear(state, action: PayloadAction<void>) {
      state.accounts = []
      state.error = undefined
    },
    enumerateAccounts(state, action: PayloadAction<void>) {
      state.step = undefined
      state.accounts = []
    },
    toggleAccount(state, action: PayloadAction<number>) {
      const index = action.payload
      state.accounts[index].selected = !state.accounts[index].selected
    },
    accountsListed(state, action: PayloadAction<LedgerAccount[]>) {
      state.accounts = action.payload
    },
    setStep(state, action: PayloadAction<LedgerStep>) {
      state.step = action.payload
    },
    operationFailed(state, action: PayloadAction<ErrorPayload>) {
      state.error = action.payload
      state.step = undefined
    },
  },
})

export const { actions: ledgerActions } = slice

export default slice.reducer
