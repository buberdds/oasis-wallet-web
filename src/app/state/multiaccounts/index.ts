import { PayloadAction } from '@reduxjs/toolkit'
import { ErrorPayload } from 'types/errors'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { MultiAccountsListAccount, MultiAccountsState, MultiAccountsStep } from './types'

export const initialState: MultiAccountsState = { accounts: [] }

const slice = createSlice({
  name: 'multiAccounts',
  initialState,
  reducers: {
    clear(state, action: PayloadAction<void>) {
      state.accounts = []
      state.error = undefined
      state.mnemonic = undefined
      state.step = undefined
    },
    enumerateAccountsFromLedger(state, action: PayloadAction<void>) {
      state.step = undefined
      state.accounts = []
    },
    enumerateAccountsFromMnemonic(state, action: PayloadAction<string>) {
      state.step = undefined
      state.accounts = []
    },
    toggleAccount(state, action: PayloadAction<number>) {
      const index = action.payload
      state.accounts[index].selected = !state.accounts[index].selected
    },
    accountsListed(state, action: PayloadAction<MultiAccountsListAccount[]>) {
      state.accounts = action.payload
    },
    setMnemonic(state, action: PayloadAction<string>) {
      state.mnemonic = action.payload
    },
    setStep(state, action: PayloadAction<MultiAccountsStep>) {
      state.step = action.payload
    },
    operationFailed(state, action: PayloadAction<ErrorPayload>) {
      state.error = action.payload
      state.mnemonic = undefined
      state.step = undefined
    },
  },
})

export const { actions: multiAccountsActions } = slice

export default slice.reducer
