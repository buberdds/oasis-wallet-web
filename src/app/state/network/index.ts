import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'
import { NetworkState, NetworkType } from './types'

export const initialState: NetworkState = {
  ticker: '',
  chainContext: '',
  selectedNetwork:
    process.env.NODE_ENV && process.env.NODE_ENV !== 'production' && !process.env.REACT_APP_BYPASS_LOCAL
      ? 'local'
      : 'mainnet',
  epoch: 0,
  minimumStakingAmount: 0,
}

const slice = createSlice({
  name: 'network',
  initialState,
  reducers: {
    selectNetwork(state, action: PayloadAction<NetworkType>) {},
    networkSelected(state, action: PayloadAction<NetworkState>) {
      Object.assign(state, action.payload)
    },
  },
})

export const { actions: networkActions } = slice

export default slice.reducer
