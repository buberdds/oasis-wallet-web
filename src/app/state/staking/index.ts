import { PayloadAction } from '@reduxjs/toolkit'
import { createSlice } from 'utils/@reduxjs/toolkit'

import { DebondingDelegation, Delegation, StakingState, Validators, ValidatorDetails } from './types'

export const initialState: StakingState = {
  debondingDelegations: [],
  delegations: [],
  validators: null,
  updateValidatorsError: null,
  selectedValidatorDetails: null,
  selectedValidator: null,
  loading: false,
}

const slice = createSlice({
  name: 'staking',
  initialState,
  reducers: {
    setLoading(state, action: PayloadAction<boolean>) {
      state.loading = action.payload
    },
    fetchAccount(state, action: PayloadAction<string>) {},
    updateValidators(state, action: PayloadAction<Validators>) {
      state.updateValidatorsError = null
      state.validators = action.payload
    },
    updateValidatorsError(state, action: PayloadAction<{ error: string; validators: Validators }>) {
      state.updateValidatorsError = action.payload.error
      state.validators = action.payload.validators
    },
    updateDelegations(state, action: PayloadAction<Delegation[]>) {
      state.delegations = action.payload
    },
    updateDebondingDelegations(state, action: PayloadAction<DebondingDelegation[]>) {
      state.debondingDelegations = action.payload
    },
    validatorSelected(state, action: PayloadAction<string>) {
      state.selectedValidator = action.payload
      state.selectedValidatorDetails = null
    },
    validatorDeselected(state, action: PayloadAction<void>) {
      state.selectedValidator = null
      state.selectedValidatorDetails = null
    },
    updateValidatorDetails(state, action: PayloadAction<ValidatorDetails>) {
      state.selectedValidatorDetails = action.payload
    },
  },
})

export const { actions: stakingActions, reducer: stakingReducer } = slice

export default slice.reducer
