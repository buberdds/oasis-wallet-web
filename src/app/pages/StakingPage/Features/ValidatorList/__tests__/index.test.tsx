import { render, screen, waitFor } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { stakingActions } from 'app/state/staking'
import * as React from 'react'
import { Provider } from 'react-redux'
import { configureAppStore } from 'store/configureStore'

import { ValidatorList } from '..'

const renderComponent = (store: any) =>
  render(
    <Provider store={store}>
      <ValidatorList />
    </Provider>,
  )

describe('<ValidatorList  />', () => {
  let store: ReturnType<typeof configureAppStore>

  beforeEach(() => {
    store = configureAppStore()
  })

  it('should match snapshot', () => {
    const component = renderComponent(store)
    store.dispatch(stakingActions.updateValidators([]))
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it('should display validator details on click', async () => {
    renderComponent(store)
    store.dispatch(
      stakingActions.updateValidators([
        {
          address: 'oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz',
          current_rate: 0.1,
          rank: 0,
          status: 'active',
          escrow: 1000,
          name: 'test-validator',
          media: {
            email_address: 'test@test.com',
            tg_chat: 'telegram',
            twitter_acc: 'https://twitter.com/my_twitter',
            website_link: 'https://test.com',
          },
        },
      ]),
    )

    let row = screen.getByText(/test-validator/)
    expect(row).toBeVisible()
    userEvent.click(row)

    const details = await screen.findByTestId('validator-item')
    row = screen.getAllByText(/test-validator/)[0]

    expect(details).toBeVisible()
    userEvent.click(row)
    await waitFor(() => expect(details).not.toBeVisible())
  })

  it('should only display the details of a single validator', async () => {
    renderComponent(store)
    store.dispatch(
      stakingActions.updateValidators([
        {
          address: 'oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz',
          current_rate: 0.1,
          rank: 0,
          status: 'active',
          escrow: 1000,
          name: 'test-validator1',
          media: {
            email_address: 'test@test.com',
            tg_chat: 'telegram',
            twitter_acc: 'https://twitter.com/my_twitter',
            website_link: 'https://test.com',
          },
        },
        {
          address: 'oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe',
          current_rate: 0.2,
          rank: 1,
          status: 'inactive',
          escrow: 1000,
          name: 'test-validator2',
          media: {
            email_address: 'test@test.com',
            tg_chat: 'telegram',
            twitter_acc: 'https://twitter.com/my_twitter',
            website_link: 'https://test.com',
          },
        },
      ]),
    )

    let row = screen.getByText(/test-validator1/)
    expect(row).toBeVisible()
    userEvent.click(row)

    let details = await screen.findByTestId('validator-item')
    expect(screen.getByTestId('validator-item-name').textContent).toEqual('test-validator1')
    expect(details).toBeVisible()

    row = screen.getByText(/test-validator2/)
    userEvent.click(row)

    details = screen.getByTestId('validator-item')
    await waitFor(() =>
      expect(screen.getByTestId('validator-item-name').textContent).toEqual('test-validator2'),
    )
    expect(details).toBeVisible()
  })
})
