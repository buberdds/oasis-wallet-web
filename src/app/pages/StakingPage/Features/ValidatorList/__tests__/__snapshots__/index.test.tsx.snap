// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidatorList  /> empty should match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c3 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  color: false;
  background-color: false;
}

.c1 {
  position: relative;
  width: 100%;
  border-radius: inherit;
  overflow-x: auto;
  overflow-y: hidden;
  min-height: 0;
}

.c2 {
  position: relative;
  width: 100%;
  display: table;
}

.c4 {
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: false;
  background-color: false;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 12px;
  }
}

<div
  class="mocked-framer-motion-div"
>
  <div
    class="c0"
  >
    common.validators
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3 rdt_Table"
          role="table"
        >
          <div
            class="c4"
          >
            <div
              style="padding: 24px;"
            >
              There are no records to display
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ValidatorList  /> list should match snapshot 1`] = `
.c10 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #666666;
  stroke: #666666;
}

.c10 g {
  fill: inherit;
  stroke: inherit;
}

.c10 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c10 *[stroke*="#"],
.c10 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c10 *[fill-rule],
.c10 *[FILL-RULE],
.c10 *[fill*="#"],
.c10 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c17 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #00C781;
  stroke: #00C781;
}

.c17 g {
  fill: inherit;
  stroke: inherit;
}

.c17 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c17 *[stroke*="#"],
.c17 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c17 *[fill-rule],
.c17 *[FILL-RULE],
.c17 *[fill*="#"],
.c17 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c19 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #FF4040;
  stroke: #FF4040;
}

.c19 g {
  fill: inherit;
  stroke: inherit;
}

.c19 *:not([stroke])[fill="none"] {
  stroke-width: 0;
}

.c19 *[stroke*="#"],
.c19 *[STROKE*="#"] {
  stroke: inherit;
  fill: none;
}

.c19 *[fill-rule],
.c19 *[FILL-RULE],
.c19 *[fill*="#"],
.c19 *[FILL*="#"] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 24px;
}

.c18 {
  margin-left: 3px;
  font-size: 18px;
  line-height: 24px;
}

.c3 {
  position: relative;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  max-width: 100%;
  color: false;
  background-color: false;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  width: 100%;
  background-color: false;
  min-height: 56px;
  border-bottom-width: 1px;
  border-bottom-color: #AAAAAAaa;
  border-bottom-style: solid;
}

.c6 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  font-size: 18px;
  font-weight: 500;
  color: false;
  padding-left: 16px;
  padding-right: 16px;
}

.c15 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  line-height: normal;
  padding-left: 16px;
  padding-right: 16px;
  word-break: break-word;
}

.c7 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 34px;
  max-width: 34px;
}

.c8 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
}

.c11 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
}

.c12 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  max-width: 100%;
  min-width: 100px;
  min-width: 110px;
  max-width: 110px;
}

.c16 {
  font-weight: 400;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  width: 100%;
  box-sizing: border-box;
  font-size: 16x;
  color: false;
  background-color: false;
  min-height: 48px;
}

.c14:not(:last-of-type) {
  border-bottom-style: solid;
  border-bottom-width: 1px;
  border-bottom-color: #AAAAAAaa;
}

.c14:hover {
  color: false;
  background-color: #88888833;
  -webkit-transition-duration: 0.15s;
  transition-duration: 0.15s;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  border-bottom-color: false;
  outline-style: solid;
  outline-width: 1px;
  outline-color: false;
}

.c14:hover {
  cursor: pointer;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  line-height: 1;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c9:focus {
  outline: none;
  color: false;
}

.c9:hover {
  color: false;
}

.c9 span.__rdt_custom_sort_icon__ i,
.c9 span.__rdt_custom_sort_icon__ svg {
  opacity: 0;
  color: inherit;
  font-size: 18px !important;
  height: 18px !important;
  width: 18px !important;
  -webkit-backface-visibility: hidden;
  backface-visibility: hidden;
  -webkit-transform-style: preserve-3d;
  -ms-transform-style: preserve-3d;
  transform-style: preserve-3d;
  -webkit-transition-duration: 125ms;
  transition-duration: 125ms;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
}

.c9 span.__rdt_custom_sort_icon__.asc i,
.c9 span.__rdt_custom_sort_icon__.asc svg {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.c9:hover {
  cursor: pointer;
}

.c9:hover span,
.c9:hover span.__rdt_custom_sort_icon__ * {
  opacity: 1;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  position: relative;
  width: 100%;
  border-radius: inherit;
  overflow-x: auto;
  overflow-y: hidden;
  min-height: 0;
}

.c2 {
  position: relative;
  width: 100%;
  display: table;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 12px;
  }
}

@media screen and (max-width:599px) {
  .c11 {
    display: none;
  }
}

@media screen and (max-width:599px) {
  .c12 {
    display: none;
  }
}

<div
  class="mocked-framer-motion-div"
>
  <div
    class="c0"
  >
    common.validators
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3 rdt_Table"
          role="table"
        >
          <div
            class="c4 rdt_TableHead"
            role="rowgroup"
          >
            <div
              class="c5 rdt_TableHeadRow"
              role="row"
            >
              <div
                class="c6 c7 rdt_TableCol"
              />
              <div
                class="c6 c7 rdt_TableCol"
              />
              <div
                class="c6 c8 rdt_TableCol"
              >
                <div
                  class="c9 rdt_TableCol_Sortable"
                  id="column-name"
                  role="columnheader"
                  tabindex="0"
                >
                  <div>
                    validator.name
                  </div>
                  <span
                    class="asc __rdt_custom_sort_icon__"
                  >
                    <svg
                      aria-label="Down"
                      class="c10"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m2 8.35 10.173 9.823L21.997 8"
                        fill="none"
                        stroke="#000"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <div
                class="c6 c11 rdt_TableCol"
              >
                <div
                  class="c9 rdt_TableCol_Sortable"
                  id="column-escrow"
                  role="columnheader"
                  tabindex="0"
                >
                  <div>
                    validator.escrow
                  </div>
                  <span
                    class="asc __rdt_custom_sort_icon__"
                  >
                    <svg
                      aria-label="Down"
                      class="c10"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m2 8.35 10.173 9.823L21.997 8"
                        fill="none"
                        stroke="#000"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </div>
              </div>
              <div
                class="c6 c12 rdt_TableCol"
              >
                <div
                  class="c9 rdt_TableCol_Sortable"
                  id="column-fee"
                  role="columnheader"
                  tabindex="0"
                >
                  <div>
                    validator.fee
                  </div>
                  <span
                    class="asc __rdt_custom_sort_icon__"
                  >
                    <svg
                      aria-label="Down"
                      class="c10"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="m2 8.35 10.173 9.823L21.997 8"
                        fill="none"
                        stroke="#000"
                        stroke-width="2"
                      />
                    </svg>
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="c13 rdt_TableBody"
            offset="250px"
            role="rowgroup"
          >
            <div
              class="c14 rdt_TableRow"
              id="row-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
              role="row"
            >
              <div
                class="c15 c7 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-icon-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
                role="gridcell"
              />
              <div
                class="c15 c7 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-status-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
                role="gridcell"
              >
                <svg
                  aria-label="Status is okay"
                  class="c17"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 22c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zM7 12l4 3 5-7"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div
                class="c15 c8 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-name-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
                role="gridcell"
              >
                test-validator1
              </div>
              <div
                class="c15 c11 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-escrow-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
                role="gridcell"
              >
                <span>
                  0
                  <span
                    class="c18"
                  >
                    TEST
                  </span>
                </span>
              </div>
              <div
                class="c15 c12 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-fee-oasis1qpc4ze5zzq3aa5mu5ttu4ku4ctp5t6x0asemymfz"
                role="gridcell"
              >
                10%
              </div>
            </div>
            <div
              class="c14 rdt_TableRow"
              id="row-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
              role="row"
            >
              <div
                class="c15 c7 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-icon-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
                role="gridcell"
              />
              <div
                class="c15 c7 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-status-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
                role="gridcell"
              >
                <svg
                  aria-label="Status is critical"
                  class="c19"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12.703 2.703a.99.99 0 0 0-1.406 0l-8.594 8.594a.99.99 0 0 0 0 1.406l8.594 8.594a.99.99 0 0 0 1.406 0l8.594-8.594a.99.99 0 0 0 0-1.406l-8.594-8.594zM8.983 14.7 14.7 8.983m-5.717 0L14.7 14.7"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div
                class="c15 c8 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-name-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
                role="gridcell"
              >
                test-validator2
              </div>
              <div
                class="c15 c11 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-escrow-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
                role="gridcell"
              >
                <span>
                  0
                  <span
                    class="c18"
                  >
                    TEST
                  </span>
                </span>
              </div>
              <div
                class="c15 c12 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-fee-oasis1qzyqaxestzlum26e2vdgvkerm6d9qgdp7gh2pxqe"
                role="gridcell"
              >
                20%
              </div>
            </div>
            <div
              class="c14 rdt_TableRow"
              id="row-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
              role="row"
            >
              <div
                class="c15 c7 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-icon-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
                role="gridcell"
              />
              <div
                class="c15 c7 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-status-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
                role="gridcell"
              >
                <svg
                  aria-label="Status is unknown"
                  class="c19"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M2 3.99C2 2.892 2.898 2 3.99 2h16.02C21.108 2 22 2.898 22 3.99v16.02c0 1.099-.898 1.99-1.99 1.99H3.99A1.995 1.995 0 0 1 2 20.01V3.99zM12 15v-1c0-1 0-1.5 1-2s2-1 2-2.5c0-1-1-2.5-3-2.5s-3 1.264-3 3m3 6v2"
                    fill="none"
                    stroke="#000"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div
                class="c15 c8 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-name-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
                role="gridcell"
              >
                test-validator3
              </div>
              <div
                class="c15 c11 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-escrow-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
                role="gridcell"
              >
                <span>
                  0
                  <span
                    class="c18"
                  >
                    TEST
                  </span>
                </span>
              </div>
              <div
                class="c15 c12 c16 rdt_TableCell"
                data-tag="allowRowEvents"
                id="cell-fee-oasis1qrfe9n26nq3t6vc9hlu9gnupwf4rm6wr0uglh3r7"
                role="gridcell"
              >
                20%
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
